### PR TITLE
LCD option to unload new filament on M600 (issue #2415)

### PR DESCRIPTION
**Pull request by Panayiotis-git**
I have met sometimes the situation described at the issue #2415 and I think that it would be useful to add the option.

Since the screen was already filled with the header and the three choices, I assigned the "Yes" choice to the header. So the new screen is:
```
>Changed correctly
 Filament not loaded
 Color not correct
 Unload filament
```

### CODE REVIEW

**Comment by Panayiotis-git** [comment]
The translations were done using the Google translate.

**Comment by Panayiotis-git** [comment]
Completed the non-MMU functionality testing, with enabled/disabled autoloading.

**Comment by Panayiotis-git** [comment]
I've rebased the code to the latest MK3 branch.

This PR also addresses two minor issues:

1. If the answer to the question "Was filament unload successful?" is "No", then the printhead moves to the left in order to make accessible the filament port. During the head movement the screen is supposed to show the message MSG_PLEASE_WAIT. But instead, it returns to the status screen.

2. Also, after unloading the filament and selecting to start the next filament loading, only for the first time, instead of showing the message "loading filament" the status screen is shown and the progress bar at the bottom.

I tested the PR with plain MK3S. No side-effects are expected to the MMU functionality, because of the code structure.

File: Firmware/Marlin_main.cpp [hunk 1]
```diff
@@ -3435,28 +3435,31 @@ static void gcode_M600(const bool automatic, const float x_position, const float
...
         {
             KEEPALIVE_STATE(PAUSED_FOR_USER);
             uint8_t choice =
                 lcd_show_fullscreen_message_yes_no_and_wait_P(_i("Was filament unload successful?"), false, LCD_LEFT_BUTTON_CHOICE); ////MSG_UNLOAD_SUCCESSFUL c=20 r=3
+            lcd_update_enable(false);
             if (choice == LCD_MIDDLE_BUTTON_CHOICE) {
                 lcd_clear();
                 lcd_puts_at_P(0, 2, _T(MSG_PLEASE_WAIT));
-                current_position[X_AXIS] -= 100;
+                current_position[X_AXIS] = 100;
```
Comment (3d-gussner) [inline Firmware/Marlin_main.cpp:3462]: Why did you change that?

Comment (Panayiotis-git) [reply Firmware/Marlin_main.cpp:3462]: If I repeatedly choose the No answer at the question "Was filament unload successful?", the print head moves to the left again and again, till it crashes to the left edge.

Another solution could be to check if the print head is at the far right and the filament port is not accessible, only then move 100 units to the left.

But then why not always move the head to a fixed position to the left?

**Comment by 3d-gussner** [comment]
I have been testing your PR and there are few changes I would like to address.

It works well and I would like to approve it with the changes.

I can make a Pr later in the afternoon or here the diff file if you want to do it yourself. The diff is based on MK3 branch with hash https://github.com/prusa3d/Prusa-Firmware/commit/02a5228f5e3034f5b9dcf9b201eb73c1341b9ee0 so a `git apply` may not work correctly.
Updated the message lengths, po files and removed some white spaces 
- `MSG_CORRECTLY c=19`
- `MSG_UNLOAD_FILAMENT c=18` as it is also used in a SUBMENU which are limited to 18 chars
   
[MK3_M600_Issue_2415_few_changes.diff.txt](https://github.com/prusa3d/Prusa-Firmware/files/13523227/MK3_M600_Issue_2415_few_changes.diff.txt)


Thanks again for a very well prepared PR.

**Comment by Panayiotis-git** [comment]
Hi @3d-gussner, thank you for your comments.
`git apply` worked perfectly!

**Review by 3d-gussner — APPROVED** [review]
Tested on MK404
image

File: Firmware/Marlin.h [hunk 2]
```diff
@@ -446,7 +446,7 @@ void gcode_M701(float fastLoadLength, uint8_t mmuSlotIndex);
 void M600_load_filament();
 void M600_load_filament_movements();
 void M600_wait_for_user();
-void M600_check_state();
+boolean M600_check_state_and_repeat();
```
Comment (DRracer) [inline Firmware/Marlin.h]: please use C++ standard `bool`

File: Firmware/Marlin_main.cpp [hunk 3]
```diff
@@ -3435,28 +3435,31 @@ static void gcode_M600(const bool automatic, const float x_position, const float
...
-        eject_slot = MMU2::mmu2.get_current_tool();
-        mmu_M600_unload_filament();
-    } else {
-        // Beep, manage nozzle heater and wait for user to start unload filament
-        M600_wait_for_user();
-        unload_filament(e_shift_late);
-    }
-    st_synchronize();          // finish moves
-    {
+    boolean repeat = false;
```
Comment (DRracer) [inline Firmware/Marlin_main.cpp]: `bool`

File: Firmware/Marlin.h (hunk 2 above)
Comment (Panayiotis-git) [reply Firmware/Marlin.h]: Strong Java influence...
